Standalone compatibility with Tractus-X Connector

Hi! I am trying to use the extension as standalone with BaSyx and Tractus-X Connector (v0.12.1). As I understood, it should be working, because Tractus-X Connector implement v3 Management-API. However, the extensions cannot be started probably due to the differenet policy context in Tractus-X. 

Here are the Exceptions I am getting:

```
WARNING 2026-06-15T13:32:21.383515132 Error expanding JSON-LD structure
JsonLdError[code=There was a problem encountered loading a remote context [code=LOADING_REMOTE_CONTEXT_FAILED]., message=There was a problem encountered loading a remote context [https://w3id.org/catenax/2025/9/policy/context.jsonld]]
        at com.apicatalog.jsonld.context.ActiveContextBuilder.fetch(ActiveContextBuilder.java:533)
        at com.apicatalog.jsonld.context.ActiveContextBuilder.create(ActiveContextBuilder.java:172)
        at com.apicatalog.jsonld.expansion.ObjectExpansion.initLocalContext(ObjectExpansion.java:197)
        at com.apicatalog.jsonld.expansion.ObjectExpansion.expand(ObjectExpansion.java:101)
        at com.apicatalog.jsonld.expansion.Expansion.compute(Expansion.java:119)
        at com.apicatalog.jsonld.processor.ExpansionProcessor.expand(ExpansionProcessor.java:124)
        at com.apicatalog.jsonld.api.ExpansionApi.get(ExpansionApi.java:132)
        at org.eclipse.edc.jsonld.TitaniumJsonLd.expand(TitaniumJsonLd.java:92)
        at de.fraunhofer.iosb.edc.remote.transform.Codec.deserialize(Codec.java:72)
        at de.fraunhofer.iosb.edc.remote.transform.Codec.lambda$deserializeList$0(Codec.java:52)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.base/java.util.Iterator.forEachRemaining(Unknown Source)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.toArray(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.toArray(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.toList(Unknown Source)
        at de.fraunhofer.iosb.edc.remote.transform.Codec.deserializeList(Codec.java:53)
        at de.fraunhofer.iosb.edc.remote.stores.ControlPlaneConnectionHandler.queryEntities(ControlPlaneConnectionHandler.java:84)
        at de.fraunhofer.iosb.edc.remote.stores.contract.RemoteContractDefinitionStore.findAll(RemoteContractDefinitionStore.java:44)
        at de.fraunhofer.iosb.app.edc.contract.ContractDefinitionService.findContracts(ContractDefinitionService.java:196)
        at de.fraunhofer.iosb.app.edc.contract.ContractDefinitionService.addToContractDefinition(ContractDefinitionService.java:94)
        at de.fraunhofer.iosb.app.handler.edc.EdcStoreHandler.register(EdcStoreHandler.java:61)
        at de.fraunhofer.iosb.app.handler.aas.AasHandler.registerSingle(AasHandler.java:173)
        at de.fraunhofer.iosb.app.handler.aas.AasHandler.lambda$initialize$2(AasHandler.java:163)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
        at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
        at de.fraunhofer.iosb.app.handler.aas.AasHandler.initialize(AasHandler.java:164)
        at de.fraunhofer.iosb.app.handler.aas.RemoteAasHandler.<init>(RemoteAasHandler.java:49)
        at de.fraunhofer.iosb.app.handler.aas.repository.period.impl.RemoteAasRepositoryHandler.<init>(RemoteAasRepositoryHandler.java:45)
        at de.fraunhofer.iosb.app.controller.RepositoryController.register(RepositoryController.java:132)
        at de.fraunhofer.iosb.app.AasExtension.bootstrapRepositories(AasExtension.java:148)
        at de.fraunhofer.iosb.app.AasExtension.start(AasExtension.java:128)
        at org.eclipse.edc.boot.system.injection.lifecycle.ExtensionLifecycleManager.bootServiceExtensions(ExtensionLifecycleManager.java:73)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.bootExtensions(BaseRuntime.java:160)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:104)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:79)
Caused by: JsonLdError[code=The document could not be loaded or parsed [code=LOADING_DOCUMENT_FAILED]., message=URL scheme [https] is not supported.]
        at com.apicatalog.jsonld.loader.SchemeRouter.loadDocument(SchemeRouter.java:59)
        at org.eclipse.edc.jsonld.TitaniumJsonLd$CachedDocumentLoader.loadDocument(TitaniumJsonLd.java:234)
        at com.apicatalog.jsonld.context.ActiveContextBuilder.fetch(ActiveContextBuilder.java:529)
        ... 46 more
WARNING 2026-06-15T13:32:21.386811879 Failed expanding json-ld Failure: There was a problem encountered loading a remote context [https://w3id.org/catenax/2025/9/policy/context.jsonld]
SEVERE 2026-06-15T13:32:21.387697831 Error booting runtime: Cannot invoke "java.util.List.stream()" because the return value of "org.eclipse.edc.spi.result.Result.getContent()" is null
java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because the return value of "org.eclipse.edc.spi.result.Result.getContent()" is null
        at de.fraunhofer.iosb.edc.remote.stores.ControlPlaneConnectionHandler.queryEntities(ControlPlaneConnectionHandler.java:90)
        at de.fraunhofer.iosb.edc.remote.stores.contract.RemoteContractDefinitionStore.findAll(RemoteContractDefinitionStore.java:44)
        at de.fraunhofer.iosb.app.edc.contract.ContractDefinitionService.findContracts(ContractDefinitionService.java:196)
        at de.fraunhofer.iosb.app.edc.contract.ContractDefinitionService.addToContractDefinition(ContractDefinitionService.java:94)
        at de.fraunhofer.iosb.app.handler.edc.EdcStoreHandler.register(EdcStoreHandler.java:61)
        at de.fraunhofer.iosb.app.handler.aas.AasHandler.registerSingle(AasHandler.java:173)
        at de.fraunhofer.iosb.app.handler.aas.AasHandler.lambda$initialize$2(AasHandler.java:163)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
        at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
        at de.fraunhofer.iosb.app.handler.aas.AasHandler.initialize(AasHandler.java:164)
        at de.fraunhofer.iosb.app.handler.aas.RemoteAasHandler.<init>(RemoteAasHandler.java:49)
        at de.fraunhofer.iosb.app.handler.aas.repository.period.impl.RemoteAasRepositoryHandler.<init>(RemoteAasRepositoryHandler.java:45)
        at de.fraunhofer.iosb.app.controller.RepositoryController.register(RepositoryController.java:132)
        at de.fraunhofer.iosb.app.AasExtension.bootstrapRepositories(AasExtension.java:148)
        at de.fraunhofer.iosb.app.AasExtension.start(AasExtension.java:128)
        at org.eclipse.edc.boot.system.injection.lifecycle.ExtensionLifecycleManager.bootServiceExtensions(ExtensionLifecycleManager.java:73)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.bootExtensions(BaseRuntime.java:160)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:104)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:79)
Exception in thread "main" org.eclipse.edc.spi.EdcException: java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because the return value of "org.eclipse.edc.spi.result.Result.getContent()" is null
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.onError(BaseRuntime.java:150)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:119)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:79)
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because the return value of "org.eclipse.edc.spi.result.Result.getContent()" is null
        at de.fraunhofer.iosb.edc.remote.stores.ControlPlaneConnectionHandler.queryEntities(ControlPlaneConnectionHandler.java:90)
        at de.fraunhofer.iosb.edc.remote.stores.contract.RemoteContractDefinitionStore.findAll(RemoteContractDefinitionStore.java:44)
        at de.fraunhofer.iosb.app.edc.contract.ContractDefinitionService.findContracts(ContractDefinitionService.java:196)
        at de.fraunhofer.iosb.app.edc.contract.ContractDefinitionService.addToContractDefinition(ContractDefinitionService.java:94)
        at de.fraunhofer.iosb.app.handler.edc.EdcStoreHandler.register(EdcStoreHandler.java:61)
        at de.fraunhofer.iosb.app.handler.aas.AasHandler.registerSingle(AasHandler.java:173)
        at de.fraunhofer.iosb.app.handler.aas.AasHandler.lambda$initialize$2(AasHandler.java:163)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
        at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
        at de.fraunhofer.iosb.app.handler.aas.AasHandler.initialize(AasHandler.java:164)
        at de.fraunhofer.iosb.app.handler.aas.RemoteAasHandler.<init>(RemoteAasHandler.java:49)
        at de.fraunhofer.iosb.app.handler.aas.repository.period.impl.RemoteAasRepositoryHandler.<init>(RemoteAasRepositoryHandler.java:45)
        at de.fraunhofer.iosb.app.controller.RepositoryController.register(RepositoryController.java:132)
        at de.fraunhofer.iosb.app.AasExtension.bootstrapRepositories(AasExtension.java:148)
        at de.fraunhofer.iosb.app.AasExtension.start(AasExtension.java:128)
        at org.eclipse.edc.boot.system.injection.lifecycle.ExtensionLifecycleManager.bootServiceExtensions(ExtensionLifecycleManager.java:73)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.bootExtensions(BaseRuntime.java:160)
        at  #org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:104)
        ... 1 more
```

The extensions fails when trying to deserialize the following json:

```json
[
  {
    "@id": "031bc996-7b78-47c1-a568-31e6badded09",
    "@type": "ContractDefinition",
    "accessPolicyId": "allow_all",
    "contractPolicyId": "allow_all",
    "assetsSelector": {
      "@type": "Criterion",
      "operandLeft": "https://w3id.org/edc/v0.0.1/ns/id",
      "operator": "in",
      "operandRight": "497239259"
    },
    "privateProperties": {
      "creator": "EDC4AAS Extension"
    },
    "@context": [
      "https://w3id.org/catenax/2025/9/policy/context.jsonld",
      {
        "tx-auth": "https://w3id.org/tractusx/auth/",
        "@vocab": "https://w3id.org/edc/v0.0.1/ns/",
        "edc": "https://w3id.org/edc/v0.0.1/ns/",
        "odrl": "http://www.w3.org/ns/odrl/2/"
      }
    ]
  }
]
```

Not quite the same error as in the first post, but I believe it still belongs to this issue. 

## Steps to Reproduce

[policy_error_reproduction.zip](https://github.com/user-attachments/files/29707863/policy_error_reproduction.zip)

1. `git clone https://github.com/FraunhoferIOSB/EDC-Extension-for-AAS`
2. `cd EDC-Extension-for-AAS/`
3. `./gradlew clean launchers:standalone:dockerize`
4. Unpack the attached archive and `cd` in the folder
5. `docker-compose up -d`
6. `docker logs iosb`
7. See the error message:
```
#SEVERE 2026-07-06T14:35:29.51483089 BAD_REQUEST: [{"message":"Policy must contain at least one permission, obligation, or prohibition","type":"ValidationFailure","path":"","invalidValue":null},{"message":"Usage policy permission must include at least the following constraints [https://w3id.org/catenax/2025/9/policy/FrameworkAgreement, https://w3id.org/catenax/2025/9/policy/UsagePurpose]","type":"ValidationFailure","path":"","invalidValue":null}]
SEVERE 2026-07-06T14:35:29.515393275 Error booting runtime: [{"message":"Policy must contain at least one permission, obligation, or prohibition","type":"ValidationFailure","path":"","invalidValue":null},{"message":"Usage policy permission must include at least the following constraints [https://w3id.org/catenax/2025/9/policy/FrameworkAgreement, https://w3id.org/catenax/2025/9/policy/UsagePurpose]","type":"ValidationFailure","path":"","invalidValue":null}]
java.lang.IllegalArgumentException: [{"message":"Policy must contain at least one permission, obligation, or prohibition","type":"ValidationFailure","path":"","invalidValue":null},{"message":"Usage policy permission must include at least the following constraints [https://w3id.org/catenax/2025/9/policy/FrameworkAgreement, https://w3id.org/catenax/2025/9/policy/UsagePurpose]","type":"ValidationFailure","path":"","invalidValue":null}]
        at de.fraunhofer.iosb.app.edc.policy.PolicyHelper.registerDefaultPolicies(PolicyHelper.java:67)
        at de.fraunhofer.iosb.app.AasExtension.start(AasExtension.java:132)
        at org.eclipse.edc.boot.system.injection.lifecycle.ExtensionLifecycleManager.bootServiceExtensions(ExtensionLifecycleManager.java:73)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.bootExtensions(BaseRuntime.java:160)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:104)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:79)
Exception in thread "main" org.eclipse.edc.spi.EdcException: java.lang.IllegalArgumentException: [{"message":"Policy must contain at least one permission, obligation, or prohibition","type":"ValidationFailure","path":"","invalidValue":null},{"message":"Usage policy permission must include at least the following constraints [https://w3id.org/catenax/2025/9/policy/FrameworkAgreement, https://w3id.org/catenax/2025/9/policy/UsagePurpose]","type":"ValidationFailure","path":"","invalidValue":null}]
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.onError(BaseRuntime.java:150)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:119)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:79)
Caused by: java.lang.IllegalArgumentException: [{"message":"Policy must contain at least one permission, obligation, or prohibition","type":"ValidationFailure","path":"","invalidValue":null},{"message":"Usage policy permission must include at least the following constraints [https://w3id.org/catenax/2025/9/policy/FrameworkAgreement, https://w3id.org/catenax/2025/9/policy/UsagePurpose]","type":"ValidationFailure","path":"","invalidValue":null}]
        at de.fraunhofer.iosb.app.edc.policy.PolicyHelper.registerDefaultPolicies(PolicyHelper.java:67)
        at de.fraunhofer.iosb.app.AasExtension.start(AasExtension.java:132)
        at org.eclipse.edc.boot.system.injection.lifecycle.ExtensionLifecycleManager.bootServiceExtensions(ExtensionLifecycleManager.java:73)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.bootExtensions(BaseRuntime.java:160)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:104)
        ... 1 more

```
## Expected Behavior
The default policy should be registered. 

## Additional information 
A policy, even without permission, obligation, prohibition and constraints, can be defined not causing error messages with tractus-x management API:

```
curl --request POST \
  --url http://localhost:8182/management/v3/policydefinitions \
  --header 'content-type: application/json' \
  --data '{
  "@context": {
    "@vocab": "https://w3id.org/edc/v0.0.1/ns/"
  },
  "@id": "allow_all",
  "policy": {
    "@context": "http://www.w3.org/ns/odrl.jsonld",
    "@type": "Set",
    "permission": [],
    "prohibition": [],
    "obligation": []
  }
}'
```

--- a/changelog.md
+++ b/changelog.md
@@ -6,6 +6,9 @@
 
 ### Bugfixes
 
+* Fixed loading/deserializing default access / contract policies
+* Fixed build goal by including duplicate files
+
 ### Miscellaneous
 
 ## V2.3.0

--- a/extensions/common/aas-lib/src/main/java/de/fraunhofer/iosb/aas/lib/model/PolicyBinding.java
+++ b/extensions/common/aas-lib/src/main/java/de/fraunhofer/iosb/aas/lib/model/PolicyBinding.java
@@ -25,7 +25,8 @@ import org.jetbrains.annotations.Nullable;
 
 import java.util.Objects;
 
-import static de.fraunhofer.iosb.constants.AasConstants.DEFAULT_POLICY_DEFINITION_ID;
+import static de.fraunhofer.iosb.constants.AasConstants.DEFAULT_ACCESS_POLICY_DEFINITION_ID;
+import static de.fraunhofer.iosb.constants.AasConstants.DEFAULT_USAGE_POLICY_DEFINITION_ID;
 
 
 /**
@@ -45,8 +46,8 @@ public record PolicyBinding(Reference referredElement, @JsonAlias("accessPolicyI
     public static PolicyBinding ofDefaults(Reference reference) {
         return new Builder()
                 .withReferredElement(ReferenceHelper.asString(reference))
-                .withAccessPolicyDefinitionId(DEFAULT_POLICY_DEFINITION_ID)
-                .withContractPolicyDefinitionId(DEFAULT_POLICY_DEFINITION_ID)
+                .withAccessPolicyDefinitionId(DEFAULT_ACCESS_POLICY_DEFINITION_ID)
+                .withContractPolicyDefinitionId(DEFAULT_USAGE_POLICY_DEFINITION_ID)
                 .build();
     }
 

--- a/extensions/common/constants/src/main/java/de/fraunhofer/iosb/constants/AasConstants.java
+++ b/extensions/common/constants/src/main/java/de/fraunhofer/iosb/constants/AasConstants.java
@@ -16,6 +16,7 @@
 package de.fraunhofer.iosb.constants;
 
 import java.util.Set;
+import java.util.UUID;
 
 
 public interface AasConstants {
@@ -25,7 +26,8 @@ public interface AasConstants {
 
     String EDC_SETTINGS_PREFIX = "edc.aas";
 
-    String DEFAULT_POLICY_DEFINITION_ID = "allow_all";
+    String DEFAULT_ACCESS_POLICY_DEFINITION_ID = UUID.randomUUID().toString();
+    String DEFAULT_USAGE_POLICY_DEFINITION_ID = UUID.randomUUID().toString();
 
     /**
      * The fields in this set are added as additional metadata in the catalog and self-description, unless configured

--- a/extensions/control-plane/codec/src/main/java/de/fraunhofer/iosb/codec/Codec.java
+++ b/extensions/control-plane/codec/src/main/java/de/fraunhofer/iosb/codec/Codec.java
@@ -66,7 +66,7 @@ public class Codec {
     }
 
 
-    public <T extends Entity> Result<T> deserialize(String entityJson, Class<T> type) {
+    public <T> Result<T> deserialize(String entityJson, Class<T> type) {
         var assetJsonObject = Json.createReader(new StringReader(entityJson)).readObject();
 
         var expandedResult = jsonLd.expand(assetJsonObject);

--- a/extensions/control-plane/edc-extension4aas/src/main/java/de/fraunhofer/iosb/app/AasExtension.java
+++ b/extensions/control-plane/edc-extension4aas/src/main/java/de/fraunhofer/iosb/app/AasExtension.java
@@ -29,6 +29,7 @@ import de.fraunhofer.iosb.app.handler.edc.EdcStoreHandler;
 import de.fraunhofer.iosb.app.model.configuration.Configuration;
 import de.fraunhofer.iosb.app.stores.repository.AasServerStore;
 import de.fraunhofer.iosb.client.exception.UnauthorizedException;
+import de.fraunhofer.iosb.codec.Codec;
 import org.eclipse.edc.connector.controlplane.asset.spi.index.AssetIndex;
 import org.eclipse.edc.connector.controlplane.contract.spi.offer.store.ContractDefinitionStore;
 import org.eclipse.edc.connector.controlplane.policy.spi.store.PolicyDefinitionStore;
@@ -43,7 +44,6 @@ import org.eclipse.edc.spi.security.Vault;
 import org.eclipse.edc.spi.system.Hostname;
 import org.eclipse.edc.spi.system.ServiceExtension;
 import org.eclipse.edc.spi.system.ServiceExtensionContext;
-import org.eclipse.edc.transform.spi.TypeTransformerRegistry;
 import org.eclipse.edc.web.spi.WebService;
 
 import java.net.ConnectException;
@@ -67,14 +67,14 @@ public class AasExtension implements ServiceExtension {
 
     public static final String NAME = "EDC4AAS Extension";
 
-    @Inject
-    private TypeTransformerRegistry typeTransformerRegistry;
     @Inject // Register public endpoints
     private PublicApiManagementService publicApiManagementService;
     @Inject
     private AssetIndex assetIndex;
     @Inject
     private ContractDefinitionStore contractDefinitionStore;
+    @Inject
+    private Codec codec;
     @Inject
     private Hostname hostname;
     @Inject(required = false)
@@ -129,7 +129,7 @@ public class AasExtension implements ServiceExtension {
     @Override
     public void start() {
         try {
-            PolicyHelper.registerDefaultPolicies(typeTransformerRegistry, monitor, policyDefinitionStore, participantId.get());
+            PolicyHelper.registerDefaultPolicies(codec, monitor, policyDefinitionStore, participantId.get());
             bootstrapRepositories();
         }
         catch (UnauthorizedException e) {

--- a/extensions/control-plane/edc-extension4aas/src/main/java/de/fraunhofer/iosb/app/edc/policy/PolicyHelper.java
+++ b/extensions/control-plane/edc-extension4aas/src/main/java/de/fraunhofer/iosb/app/edc/policy/PolicyHelper.java
@@ -15,9 +15,8 @@
  */
 package de.fraunhofer.iosb.app.edc.policy;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
 import de.fraunhofer.iosb.app.model.configuration.Configuration;
-import jakarta.json.JsonObject;
+import de.fraunhofer.iosb.codec.Codec;
 import org.eclipse.edc.connector.controlplane.policy.spi.PolicyDefinition;
 import org.eclipse.edc.connector.controlplane.policy.spi.store.PolicyDefinitionStore;
 import org.eclipse.edc.policy.model.Action;
@@ -26,12 +25,13 @@ import org.eclipse.edc.policy.model.Policy;
 import org.eclipse.edc.spi.monitor.Monitor;
 import org.eclipse.edc.spi.result.Result;
 import org.eclipse.edc.spi.result.StoreResult;
-import org.eclipse.edc.transform.spi.TypeTransformerRegistry;
 
 import java.io.IOException;
+import java.nio.file.Files;
 import java.nio.file.Path;
 
-import static de.fraunhofer.iosb.constants.AasConstants.DEFAULT_POLICY_DEFINITION_ID;
+import static de.fraunhofer.iosb.constants.AasConstants.DEFAULT_ACCESS_POLICY_DEFINITION_ID;
+import static de.fraunhofer.iosb.constants.AasConstants.DEFAULT_USAGE_POLICY_DEFINITION_ID;
 import static org.eclipse.edc.jsonld.spi.PropertyAndTypeNames.ODRL_USE_ACTION_ATTRIBUTE;
 import static org.eclipse.edc.spi.result.StoreFailure.Reason.ALREADY_EXISTS;
 
@@ -44,37 +44,50 @@ public abstract class PolicyHelper {
     /**
      * Register default access and contract policies either by configuration variables or using internal (minimal) policies.
      *
-     * @param typeTransformerRegistry Deserialize policy
+     * @param codec Deserialize policy
      * @param monitor monitor to log messages.
      * @param policyDefinitionStore Store implementation to register policies.
      * @param participantId Used for the default policies.
      */
-    public static void registerDefaultPolicies(TypeTransformerRegistry typeTransformerRegistry, Monitor monitor, PolicyDefinitionStore policyDefinitionStore,
+    public static void registerDefaultPolicies(Codec codec, Monitor monitor, PolicyDefinitionStore policyDefinitionStore,
                                                String participantId) {
         Configuration configuration = Configuration.getInstance();
 
-        Policy defaultPolicy = getPolicy(typeTransformerRegistry, monitor, participantId, configuration.getDefaultAccessPolicyPath());
+        Policy defaultAccessPolicy = getPolicy(codec, monitor, participantId, configuration.getDefaultAccessPolicyPath());
+        Policy defaultUsagePolicy = getPolicy(codec, monitor, participantId, configuration.getDefaultContractPolicyPath());
 
-        var defaultPolicyDefinition = PolicyDefinition.Builder.newInstance()
-                .id(DEFAULT_POLICY_DEFINITION_ID)
-                .policy(defaultPolicy)
+        var defaultAccessPolicyDefinition = PolicyDefinition.Builder.newInstance()
+                .id(DEFAULT_ACCESS_POLICY_DEFINITION_ID)
+                .policy(defaultAccessPolicy)
                 .participantContextId(participantId)
                 .build();
 
-        StoreResult<PolicyDefinition> storeResult = policyDefinitionStore.create(defaultPolicyDefinition);
+        StoreResult<PolicyDefinition> accessPolicyDefinitionStoreResult = policyDefinitionStore.create(defaultAccessPolicyDefinition);
 
-        if (storeResult.failed() && ALREADY_EXISTS != storeResult.reason()) {
-            throw new IllegalArgumentException(storeResult.getFailureDetail());
+        if (accessPolicyDefinitionStoreResult.failed() && ALREADY_EXISTS != accessPolicyDefinitionStoreResult.reason()) {
+            throw new IllegalArgumentException(accessPolicyDefinitionStoreResult.getFailureDetail());
         }
+
+        var defaultUsagePolicyDefinition = PolicyDefinition.Builder.newInstance()
+                .id(DEFAULT_USAGE_POLICY_DEFINITION_ID)
+                .policy(defaultUsagePolicy)
+                .participantContextId(participantId)
+                .build();
+
+        StoreResult<PolicyDefinition> usagePolicyDefinitionStoreResult = policyDefinitionStore.create(defaultUsagePolicyDefinition);
+
+        if (usagePolicyDefinitionStoreResult.failed() && ALREADY_EXISTS != usagePolicyDefinitionStoreResult.reason()) {
+            throw new IllegalArgumentException(usagePolicyDefinitionStoreResult.getFailureDetail());
+        }
+
     }
 
 
-    private static Policy getPolicy(TypeTransformerRegistry typeTransformerRegistry, Monitor monitor, String participantId, String path) {
+    private static Policy getPolicy(Codec codec, Monitor monitor, String participantId, String path) {
         Policy policy;
         if (path != null) {
-            policy = getPolicyDefinitionFromFile(typeTransformerRegistry, path).orElse(failure -> {
-                monitor.severe(failure.getFailureDetail());
-                return defaultPolicy(participantId);
+            policy = getPolicyFromFile(codec, path).orElse(failure -> {
+                throw new IllegalArgumentException(String.format("Could not parse default policy from %s: %s", path, failure.getFailureDetail()));
             });
         }
         else {
@@ -84,12 +97,12 @@ public abstract class PolicyHelper {
     }
 
 
-    private static Result<Policy> getPolicyDefinitionFromFile(TypeTransformerRegistry typeTransformerRegistry, String filePath) {
+    private static Result<Policy> getPolicyFromFile(Codec codec, String filePath) {
         try {
-            return typeTransformerRegistry.transform(new ObjectMapper().readValue(Path.of(filePath).toFile(), JsonObject.class), Policy.class);
+            return codec.deserialize(Files.readString(Path.of(filePath)), Policy.class);
         }
         catch (IOException ioException) {
-            return Result.failure(String.format("Could not find a valid policy at path %s. Using internal policy as default.", filePath));
+            return Result.failure(ioException.getMessage());
         }
     }
 

--- a/extensions/control-plane/edc-extension4aas/src/main/java/de/fraunhofer/iosb/app/handler/aas/AasHandler.java
+++ b/extensions/control-plane/edc-extension4aas/src/main/java/de/fraunhofer/iosb/app/handler/aas/AasHandler.java
@@ -114,7 +114,9 @@ public abstract class AasHandler<C extends AasServerClient> {
                 .map(StoreResult::getFailureDetail)
                 .toList();
 
-        monitor.warning(String.format("Failed unregistering assets with IDs %s", unregisterFailedMessages));
+        if (unregisterFailedMessages.size() > 0) {
+            monitor.warning(String.format("Failed unregistering assets with IDs %s", unregisterFailedMessages));
+        }
 
         monitor.info(String.format("Unregistered %s AAS elements from repository %s.", filtered.size() - unregisterFailedMessages.size(),
                 client.getUri()));

--- a/extensions/edc-connector-client/src/main/java/de/fraunhofer/iosb/edc/remote/stores/policy/RemotePolicyDefinitionStore.java
+++ b/extensions/edc-connector-client/src/main/java/de/fraunhofer/iosb/edc/remote/stores/policy/RemotePolicyDefinitionStore.java
@@ -60,8 +60,8 @@ public class RemotePolicyDefinitionStore extends ControlPlaneConnectionHandler<P
         // Since you cannot create StoreResult<T> from StoreResult<U>:
         String failureDetail = createResult.getFailureDetail();
         return switch (createResult.reason()) {
-            case NOT_FOUND -> StoreResult.notFound(createResult.getFailureDetail());
-            case ALREADY_EXISTS -> StoreResult.alreadyExists(createResult.getFailureDetail());
+            case NOT_FOUND -> StoreResult.notFound(failureDetail);
+            case ALREADY_EXISTS -> StoreResult.alreadyExists(failureDetail);
             case DUPLICATE_KEYS -> StoreResult.duplicateKeys(failureDetail);
             case ALREADY_LEASED -> StoreResult.alreadyLeased(failureDetail);
             default -> StoreResult.generalError(failureDetail);

--- a/extensions/repository-aas/src/main/java/de/fraunhofer/iosb/model/context/AasServerContext.java
+++ b/extensions/repository-aas/src/main/java/de/fraunhofer/iosb/model/context/AasServerContext.java
@@ -20,7 +20,8 @@ import org.eclipse.digitaltwin.aas4j.v3.model.Reference;
 import java.net.URI;
 import java.util.Objects;
 
-import static de.fraunhofer.iosb.constants.AasConstants.DEFAULT_POLICY_DEFINITION_ID;
+import static de.fraunhofer.iosb.constants.AasConstants.DEFAULT_ACCESS_POLICY_DEFINITION_ID;
+import static de.fraunhofer.iosb.constants.AasConstants.DEFAULT_USAGE_POLICY_DEFINITION_ID;
 
 
 /**
@@ -90,8 +91,8 @@ public abstract class AasServerContext {
 
 
         protected void validate() {
-            defaultAccessPolicyDefinitionId = Objects.requireNonNullElse(defaultAccessPolicyDefinitionId, DEFAULT_POLICY_DEFINITION_ID);
-            defaultContractPolicyDefinitionId = Objects.requireNonNullElse(defaultContractPolicyDefinitionId, DEFAULT_POLICY_DEFINITION_ID);
+            defaultAccessPolicyDefinitionId = Objects.requireNonNullElse(defaultAccessPolicyDefinitionId, DEFAULT_ACCESS_POLICY_DEFINITION_ID);
+            defaultContractPolicyDefinitionId = Objects.requireNonNullElse(defaultContractPolicyDefinitionId, DEFAULT_USAGE_POLICY_DEFINITION_ID);
         }
     }
 }

--- a/samples/config/standalone.properties
+++ b/samples/config/standalone.properties
@@ -33,3 +33,5 @@ edc.web.rest.cors.enabled=true
 edc.web.rest.cors.origins=*
 edc.web.rest.cors.headers=*
 edc.web.rest.cors.methods=GET, POST, DELETE, PUT, OPTIONS
+
+edc.jsonld.https.enabled=true

--- a/system-tests/resources/contracts.json
+++ b/system-tests/resources/contracts.json
@@ -1,8 +1,6 @@
 [
   {
     "@type": "ContractDefinition",
-    "accessPolicyId": "allow_all",
-    "contractPolicyId": "allow_all",
     "assetsSelector": [
       {
         "@type": "Criterion",

--- a/system-tests/resources/policy.json
+++ b/system-tests/resources/policy.json
@@ -1,6 +1,22 @@
 [
   {
-    "@id": "allow_all",
+    "@type": "PolicyDefinition",
+    "policy": {
+      "@type": "Set",
+      "permission": [
+        {
+          "action": "use"
+        }
+      ],
+      "prohibition": [],
+      "obligation": [],
+      "odrl:assigner": "provider"
+    },
+    "@context": [
+      "https://w3id.org/edc/connector/management/v2"
+    ]
+  },
+  {
     "@type": "PolicyDefinition",
     "policy": {
       "@type": "Set",

